src/components/license-activation/data/service.js

```diff
@@ -1,7 +1,25 @@
+import Cookies from 'universal-cookie';
 import { getAuthenticatedHttpClient } from '@edx/frontend-platform/auth';
 import qs from 'query-string';
 
-export function activateLicense(activationKey) {
+export async function activateLicense(activationKey) {
+  // If the user has not refreshed their JWT since they created their account,
+  // we should refresh it so that they'll have appropriate roles (if available),
+  // and thus, have any appropriate permissions when making downstream requests.
+  const loginRefreshUrl = `${process.env.LMS_BASE_URL}/login_refresh`;
+
+  try {
+    await getAuthenticatedHttpClient().post(loginRefreshUrl);
+  } catch (error) {
+    const userIsUnauthenticated = error.response && error.response.status === 401;
```

Comment:
nit: could use the optional chaining operator (`?.`) here 😄 

`const userIsUnauthenticated = error.response?.status === 401`

also, super nit: maybe `isUserUnauthenticated` (makes it a bit clearer its a bool by the name when it starts with "is")?